pkg/validation/validation.go

```diff
@@ -134,7 +135,7 @@ func parameterRef(gc *gatewayv1alpha1.GatewayClass) error {
 }
 
 // Gateway returns an error if gw is an invalid Gateway.
```

Comment:
let's update the godoc to indicate this now returns the Contour associated with the Gateway, if the Gateway is valid.